SpokeStack/TFLiteWakewordRecognizer.swift

```diff
@@ -0,0 +1,478 @@
+                    let encodeResults = encodeOutput.data.withUnsafeBytes { (pointer: UnsafePointer<Float32>) -> [Float32] in
+                        Array<Float32>(UnsafeBufferPointer(start: pointer, count: encodeOutput.data.count / 4))}
+                    self.encodeWindow.rewind().seek(self.encodeWidth)
+                    for r in encodeResults {
+                        try self.encodeWindow.write(r)
+                        //encodeCollector.append(r)
+                    }
+                    let stateOutput = try model.output(at: Tensors.state.rawValue)
+                    let stateResults = stateOutput.data.withUnsafeBytes { (pointer: UnsafePointer<Float32>) -> [Float32] in
+                        Array<Float32>(UnsafeBufferPointer(start: pointer, count: stateOutput.data.count / 4))}
+                    for r in stateResults {
+                        try self.encodeState.write(r)
+                        //stateCollector.append(r)
+                    }
+                } else {
+                    throw WakewordModelError.encode("encode.lite was not initialized")
+                }
+            }
+        }
+    }
+    
+    private func detect() throws -> Bool {
+        if self.encodeWindow.isFull {
+            do {
+                if let model = self.detectModel {
+                    /// inputs
+                    var encodeWindowArray: Array<Float32> = []
+                    self.encodeWindow.rewind()
+                    while !self.encodeWindow.isEmpty {
+                        let f = try self.encodeWindow.read()
+                        encodeWindowArray.append(f)
+                    }
+                    _ = try encodeWindowArray
+                        .withUnsafeBytes(
+                            {try model.copy(Data($0), toInputAt: 0)})
+                    
+                    /// calculate
+                    try model.invoke()
+                    
+                    /// outputs
+                    let detectOutput = try model.output(at: 0)
+                    let detectResults = detectOutput.data.withUnsafeBytes { (pointer: UnsafePointer<Float32>) -> [Float32] in
+                        Array<Float32>(UnsafeBufferPointer(start: pointer, count: detectOutput.data.count / 4))}
+                    let posterior = detectResults[0]
+                    if posterior > self.posteriorMax {
+                        self.posteriorMax = posterior
+                    }
+                    if posterior > self.posteriorThreshold {
+                        return true
+                    }
+                 }
+            }
+        }
+        return false
+    }
+    
+    private func reset() -> Void {
+        /// Empty the sample buffer, so that only contiguous
+        /// speech samples are written to it
+        self.sampleWindow.reset()
+        
+        /// Reset and fill the other buffers,
+        /// which prevents them from lagging the detection
+        self.frameWindow.reset().fill(0)
```

Comment:
The other 3 ring buffers (encode, state, and detect) should also be reset/filled here.